test/regex_test.cpp

```diff
@@ -0,0 +1,186 @@
+#include "gtest/gtest.h"
+#include "regex_yaml.h"
+#include "stream.h"
+
+using YAML::RegEx;
+using YAML::Stream;
+
+namespace {
+TEST(RegExTest, Empty) {
+  RegEx empty;
+  EXPECT_TRUE(empty.Matches(std::string()));
+  EXPECT_EQ(0, empty.Match(std::string()));
+  for (int ii = Stream::eof() + 1; ii < 128; ++ii) {
```

Comment:
1) I'd prefer "i" to "ii"

2) Starting with Stream::eof() + 1 is probably not necessary. I'd hard-code the constant here. If you prefer, feel free to make a named constant at the top of this file (and if you do that, do one for the 128 as well.